### PR TITLE
New CustomTable as Visualization

### PR DESCRIPTION
**Pull request by mmelograno**
This feature adds a new custom visualization for queries. The custom table allows to align columns, put text in bold, italic or create a custom link. To add this feature, we used smart-table. However we added a custom-table-renderer.html to allow those custom options for each column.
Another thing to point out is that we added new features on widgets, for displaying the widget or not (depending of the parameters read for query) and to read parameters from url.

Spinner gif was added to show it when the queryResult.getStatus is waiting or processing.

### CODE REVIEW

**Comment by arikfr** [comment]
Can you please attach screenshots of the new features? (the custom table, the widget with parameters, etc).

Also, please avoid opening pull requests for more than one feature. It makes it harder to review and merge.

**Comment by arikfr** [comment]
(I mean open one pull request for one feature. Multiple pull requests are obviously encouraged)

File: rd_ui/app/scripts/directives/dashboard_directives.js [hunk 1]
```diff
@@ -123,6 +123,9 @@
         replace: true,
         link: function($scope, element, attrs) {
           $scope.widgetSizes = [{
+            name: 'Hidden',
+            value: 0
+          }, {
```
Comment (arikfr) [inline rd_ui/app/scripts/directives/dashboard_directives.js]: Check [line 151](https://github.com/mmelograno/redash/blob/master/rd_ui/app/scripts/directives/dashboard_directives.js#L151) in this file, we already add the hidden option there.

**Comment by mmelograno** [comment]
Hi Arick,

I made a pull request for covering the CustomTable Visualization. The 'Show Always' option when you create a Widget is used by the CustomTable and 'Read parameters for url' is displayed on CustomTable Link option. However, the spinning is another independant feature.

The CustomTable Visualization on a DrillDown Dashboard is displayed like this:
drilldown
You can find that the first row is a link with an href. This image shows also the redirect for 'christies' Marketing Group. You can see also that the 'Amount_of_Listings' column is aligned to the left.

On the queries page you can add a 'New Visualization' as 'Custom Table'. Below you can find an example of that:
visualization
Below, is displayed the CustomTable options. For each column you can Show/Hide, Align, Italic, Bold and Linked.
editcustomtableoptions

The Link option allow us to render a custom link on the table. The next picture show how this option is set up. Notice that the first input renders the href on column (will render ?p_mgid=##v_mgid##). This will render the v_mgid associated for each row. The second input renders the visualization on column (will render the Marketing Group associated).
Finally, the option 'Use parameters from url' will render on each link column the parameters that browser url has. For instance if I have this url on browser http://localhost:9001/dashboard/2---drill-down-by-dates?maxAge=60&p_startdate=2015-11-02&p_enddate=2015-11-03, on the link column hwill be added the parameters

**Pull request closed without merging**